src/popup.ts

```diff
@@ -284,13 +276,13 @@ const handleClearSelectionClicks = async () => {
   });
 }
 
-const setBlockExecutionStatus = (blockExecutionValue) => {
+const setBlockExecutionStatus = (blockExecutionValue: BlockExecution) => {
   waitForElement("#block-execution-toggle input").then(function (toggleInput) {
     blockExecutionValue
       ? toggleInput.checked = false
       : toggleInput.checked = true;
 
-    toggleInput.addEventListener('click', function (e) {
+    toggleInput.addEventListener('click', function (e: any) {
```

Comment:
You'll have to add `"dom"` to _tsconfig.json_ as a lib for this to work.
```suggestion
    toggleInput.addEventListener('click', function (e: MouseEvent) {
```